Source/monster.cpp

```diff
@@ -4758,69 +4758,43 @@ bool LineClearMissile(Point startPoint, Point endPoint)
 
 bool LineClear(bool (*Clear)(int, Point), int entity, Point startPoint, Point endPoint)
 {
-	int d;
-	int xincD, yincD, dincD, dincH;
-	bool done = false;
-
 	Point position = startPoint;
 
-	int dx = endPoint.x - position.x;
-	int dy = endPoint.y - position.y;
-	if (abs(dx) > abs(dy)) {
-		if (dx < 0) {
-			std::swap(position, endPoint);
-			dx = -dx;
-			dy = -dy;
-		}
-		if (dy > 0) {
-			d = 2 * dy - dx;
-			dincD = 2 * dy;
-			dincH = 2 * (dy - dx);
-			yincD = 1;
+	int d1 = endPoint.x - position.x;
+	int d2 = endPoint.y - position.y;
+
+	bool horizontal = abs(d1) > abs(d2);
+
+	if (!horizontal)
+		std::swap(d1, d2);

```

Comment:
I have no idea why this happens. Can you add comments to the method to explain what it does? Or rename it to something more specific?

Reply:
It's swapping the x/y axis in order to move down instead of to the right in the grid.

Reply:
You misunderstood my comment. I'm saying I don't understand what the whole function is doing, so it's hard to evaluate the logic as a whole (including these swaps).

It's hard to provide feedback if the initial logic was not clear.

Reply:
It checks to see if there is a line free of obstacles between two point. Often used when a monster checks to see if it can shoot at the player, but also for some of the AI route planning logic.

Reply:
Ah ok. Do you think we can include comments to the method as part of this change? Or on a previous commit?

Reply:
BTW, it puzzles me why we do any swaps of anything if the purpose is to check for a clear line?

Reply:
> Ah ok. Do you think we can include comments to the method as part of this change? Or on a previous commit?

Defiantly, I just though the functions porous was obvious :joy:

Reply:
> BTW, it puzzles me why we do any swaps of anything if the purpose is to check for a clear line?

The loop expects to walk in a positive direction with most movement being horizontal. This limits it to 1/4 of all possible directions. So in order to be able to do arbitrary moves we do up to two swaps (the original had one unrolled and an `if-else`) to flip the coordinate system in a way that fits the way the loop works.

Reply:
![image](https://user-images.githubusercontent.com/204594/120088900-a7d1d800-c0f5-11eb-8c79-8521ba7d3a08.png)